public/client.js

```diff
@@ -4,56 +4,42 @@
 $(document).ready(function () {
 	'use strict';
 
-	var intervalId = 0;
-	var pollInProgress = false;
+
+	const MAX_INTERVAL = 5000;
+	const MIN_INTERVAL = 500;
+	const USERS_PER_INTERVAL_INCREASE = 10;
+
+	let interval = MIN_INTERVAL;
 
 	$(window).on('action:ajaxify.end', function (ev, data) {
-		if (ajaxify.data.template.topic) {
+		if (ajaxify.data.template.topic && app.user.uid) {
 			renderBrowsingUsers();
-			startPolling();
-		} else {
-			stopPolling();
+			setTimeout(renderBrowsingUsers, interval);
```

Comment:
If user isn't on the topic page there is no need to render browsing users. So not sure if there is a need to do setTimeout here

Reply:
there's a check at the top of the function for topic page, should be ok I think
also tested going in/out of topic + closing browser, etc

Reply:
nvm turns out I didn't need it anyway, removed